src/riak_core_handoff_manager.erl

```diff
@@ -404,11 +404,16 @@ calc_stats(#handoff_status{stats=Stats,timestamp=StartTS,size=Size}) ->
              {pct_done_decimal, Done}]
     end.
 
+-spec get_size(db_size()|undefined) -> db_size_result()|undefined.
 get_size({F, dynamic}) ->
     F();
 get_size(S) ->
     S.
 
+-spec calc_pct_done(
+    non_neg_integer(),
+    non_neg_integer(),
+    db_size_result() | undefined) -> float() | undefined.
 calc_pct_done(_, _, undefined) ->
     undefined;
 calc_pct_done(Objs, _, {Size, objects}) ->
```

Comment:
This is not part of the refactoring and worked before... so your size is never zero when you compute it, just when you initialize it in the `handoff_status` record. If it were 0, you would devide by zero with a crash as consequence.

Reply:
Reverted the order so any unexpected result result (including a Size of 0) will result in a response of undefined.